src/main/java/edu/ie3/datamodel/graph/ImpedanceWeightedGraph.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * © 2021. TU Dortmund University,
+ * Institute of Energy Systems, Energy Efficiency and Energy Economics,
+ * Research group Distribution grid planning and operation
+*/
+package edu.ie3.datamodel.graph;
+
+import edu.ie3.datamodel.models.input.NodeInput;
+import java.util.function.Supplier;
+import javax.measure.Quantity;
+import javax.measure.quantity.ElectricResistance;
+import org.jgrapht.graph.SimpleWeightedGraph;
+import tech.units.indriya.ComparableQuantity;
+
+/** An impedance weighted graph that uses {@link ImpedanceWeightedEdge}s as edge type. */
+public class ImpedanceWeightedGraph extends SimpleWeightedGraph<NodeInput, ImpedanceWeightedEdge> {
+
+  private static final long serialVersionUID = -2797654003980753342L;
+
+  public ImpedanceWeightedGraph() {
+    super(ImpedanceWeightedEdge.class);
+  }
+
+  public ImpedanceWeightedGraph(
+      Supplier<NodeInput> vertexSupplier, Supplier<ImpedanceWeightedEdge> edgeSupplier) {
+    super(vertexSupplier, edgeSupplier);
+  }
+
+  /**
+   * Assigns a {@link Quantity} of type {@link ElectricResistance} to an instance of edge {@link
+   * ImpedanceWeightedEdge}
+   *
+   * @param edge edge whose weight should be altered
+   * @param weight the weight of the {@link ImpedanceWeightedEdge}
+   */
+  public void setWeightQuantity(
+      ImpedanceWeightedEdge edge, ComparableQuantity<ElectricResistance> weight) {
+    double weightDouble =
+        weight.to(ImpedanceWeightedEdge.DEFAULT_IMPEDANCE_UNIT).getValue().doubleValue();
+    super.setEdgeWeight(edge, weightDouble);
+  }
+
+  /**
+   * The only purpose for overriding this method is to provide a better indication of the unit that
+   * is expected to be passed in. It is highly advised to use the {@link
+   * ImpedanceWeightedGraph#setWeightQuantity(ImpedanceWeightedEdge, ComparableQuantity)} for safety
+   * purposes that the provided edge weight is correct.
+   *
+   * @param edge the edge whose weight should be altered
+   * @param impedanceInOhm the weight of the {@link ImpedanceWeightedEdge} in ohm
+   * @deprecated Use {@link ImpedanceWeightedGraph#setWeightQuantity(ImpedanceWeightedEdge,
+   *     ComparableQuantity)} instead, as it provides means for proper unit handling
+   */
+  @Override
+  @Deprecated
```

Comment:
No, it is specifically not intended to remove this override. The deprecation is ony there to indicate directly within the IDE, that this method is "deprecated to use". Why is it deprecated? Because you can set a number without caring about which unit it adheres to. Yes, we could throw an Exception, but then you only get to know, that you did not so nice stuff during the runtime of your program. The override with deprecation is here, because if it wouldn't be here, you could use the same method from super class without getting warned at all.